auth/authenticator.go

```diff
@@ -51,6 +58,16 @@ func (l *LsatAuthenticator) Accept(header *http.Header, serviceName string) bool
 		return false
 	}
 
+	// Make sure the backend has the invoice recorded as settled.
+	err = l.checker.VerifyInvoiceStatus(
```

Comment:
Just to confirm, will the timeout here block processing further requests? I believe `ServeHTTP` is spawned as a goroutine so it should be fine?

Reply:
Yeah IIRC it is indeed a distinct goroutine.

Reply:
Affirmative: https://golang.org/src/net/http/server.go#L2962